### ISSUE TITLE
an exception I met when running pgbouncer on master

### ISSUE THREAD

**Issue by kuien** [greenplum-db/gpdb#921]
We found an issue when query through pgbouncer using sql

```
psql -p 65432 postgres -t -c 'select 1=1;'
```

then GPDB goes to recovery mode with logs

```
2016-07-05 00:46:18.445540 EDT,"gpadmin","postgres",p9655,th-27006912,"[local]",,2016-07-05 00:46:18 EDT,800,con9,cmd1,seg-1,,dx4,x800,sx1,"LOG","00000","statement: select 1=1;",,,,,,"select 1=1;",0,,"postgres.c",1616,
2016-07-05 00:47:09.219254 EDT,"gpadmin","postgres",p9763,th-27006912,"127.0.0.1","35375",2016-07-05 00:47:09 EDT,801,con10,cmd1,seg-1,,dx5,x801,sx1,"LOG","00000","statement: SET application_name='psql';",,,,,,"SET application_name='psql';",0,,"postgres.c",1616,
2016-07-05 00:47:09.364553 EDT,"gpadmin","postgres",p9763,th-27006912,"127.0.0.1","35375",2016-07-05 00:47:09 EDT,802,con10,cmd1,seg-1,,dx6,x802,sx1,"FATAL","XX000","Unexpected internal error (postgres.c:886)","FailedAssertion(""!(ActiveSnapshot != ((void *)0))"", File: ""postgres.c"", Line: 886)","Process 9763 will wait for gp_debug_linger=120 seconds before termination.
Note that its locks and other resources will not be released until then.",,,,"SELECT usename, passwd FROM pg_shadow WHERE usename=$1",0,,"postgres.c",886,"Stack trace:
1    0x99c338 postgres errstart (elog.c:514)
2    0x98e9d9 postgres ExceptionalCondition (assert.c:39)
3    0x8678f1 postgres <symbol not found> (postgres.c:886)
4    0x8680ac postgres <symbol not found> (postgres.c:2552)
5    0x86b1bb postgres PostgresMain (postgres.c:5273)
6    0x7fc2f2 postgres <symbol not found> (postmaster.c:6403)
7    0x7fd539 postgres PostmasterMain (postmaster.c:1542)
8    0x48cf3f postgres main (main.c:206)
9    0x7f6afbb8cb15 libc.so.6 __libc_start_main (??:0)
10   0x48d05d postgres <symbol not found> (??:?)
"
2016-07-05 00:47:16.746134 EDT,"gpadmin","gpadmin",p9800,th-27006912,"[local]",,2016-07-05 00:47:16 EDT,0,con11,,seg-1,,,,sx1,"FATAL","3D000","database ""gpadmin"" does not exist",,,,,,,0,,"postinit.c",786,
2016-07-05 00:47:21.323129 EDT,"gpadmin","postgres",p9829,th-27006912,"[local]",,2016-07-05 00:47:21 EDT,803,con12,cmd1,seg-1,,dx7,x803,sx1,"LOG","00000","statement: select 1=1;",,,,,,"select 1=1;",0,,"postgres.c",1616,
2016-07-05 00:47:31.450409 EDT,"gpadmin","postgres",p9859,th-27006912,"127.0.0.1","35381",2016-07-05 00:47:31 EDT,804,con13,cmd1,seg-1,,dx8,x804,sx1,"LOG","00000","statement: SET application_name='psql';",,,,,,"SET application_name='psql';",0,,"postgres.c",1616,
2016-07-05 00:47:31.597947 EDT,"gpadmin","postgres",p9859,th-27006912,"127.0.0.1","35381",2016-07-05 00:47:31 EDT,805,con13,cmd1,seg-1,,dx9,x805,sx1,"FATAL","XX000","Unexpected internal error (postgres.c:886)","FailedAssertion(""!(ActiveSnapshot != ((void *)0))"", File: ""postgres.c"", Line: 886)","Process 9859 will wait for gp_debug_linger=120 seconds before termination.
Note that its locks and other resources will not be released until then.",,,,"SELECT usename, passwd FROM pg_shadow WHERE usename=$1",0,,"postgres.c",886,"Stack trace:
1    0x99c338 postgres errstart (elog.c:514)
2    0x98e9d9 postgres ExceptionalCondition (assert.c:39)
3    0x8678f1 postgres <symbol not found> (postgres.c:886)
4    0x8680ac postgres <symbol not found> (postgres.c:2552)
5    0x86b1bb postgres PostgresMain (postgres.c:5273)
6    0x7fc2f2 postgres <symbol not found> (postmaster.c:6403)
7    0x7fd539 postgres PostmasterMain (postmaster.c:1542)
8    0x48cf3f postgres main (main.c:206)
9    0x7f6afbb8cb15 libc.so.6 __libc_start_main (??:0)
10   0x48d05d postgres <symbol not found> (??:?)
"
2016-07-05 00:47:44.882166 EDT,"gpadmin","postgres",p9924,th-27006912,"127.0.0.1","35387",2016-07-05 00:47:44 EDT,806,con14,cmd1,seg-1,,dx10,x806,sx1,"LOG","00000","statement: SET application_name='psql';",,,,,,"SET application_name='psql';",0,,"postgres.c",1616,
2016-07-05 00:47:45.023879 EDT,"gpadmin","postgres",p9924,th-27006912,"127.0.0.1","35387",2016-07-05 00:47:44 EDT,807,con14,cmd1,seg-1,,dx11,x807,sx1,"FATAL","XX000","Unexpected internal error (postgres.c:886)","FailedAssertion(""!(ActiveSnapshot != ((void *)0))"", File: ""postgres.c"", Line: 886)","Process 9924 will wait for gp_debug_linger=120 seconds before termination.
Note that its locks and other resources will not be released until then.",,,,"SELECT usename, passwd FROM pg_shadow WHERE usename=$1",0,,"postgres.c",886,"Stack trace:
1    0x99c338 postgres errstart (elog.c:514)
2    0x98e9d9 postgres ExceptionalCondition (assert.c:39)
3    0x8678f1 postgres <symbol not found> (postgres.c:886)
4    0x8680ac postgres <symbol not found> (postgres.c:2552)
5    0x86b1bb postgres PostgresMain (postgres.c:5273)
6    0x7fc2f2 postgres <symbol not found> (postmaster.c:6403)
7    0x7fd539 postgres PostmasterMain (postmaster.c:1542)
8    0x48cf3f postgres main (main.c:206)
9    0x7f6afbb8cb15 libc.so.6 __libc_start_main (??:0)
10   0x48d05d postgres <symbol not found> (??:?)
"
2016-07-05 00:48:21.377956 EDT,"gpadmin","postgres",p10007,th-27006912,"127.0.0.1","35397",2016-07-05 00:48:21 EDT,808,con15,cmd1,seg-1,,dx12,x808,sx1,"LOG","00000","statement: SET application_name='psql';",,,,,,"SET application_name='psql';",0,,"postgres.c",1616,
```

the version of GPDB is:

```
○ → psql postgres
psql (8.3.23)
Type "help" for help.

postgres=# select version();
                                                                                                  version

----------------------------------------------------------------------------------------------------------------------------------------
--------------------------------------------------------------------
 PostgreSQL 8.3.23 (Greenplum Database 4.3.99.00 build dev) on x86_64-unknown-linux-gnu, compiled by GCC gcc (GCC) 4.8.5 20150623 (Red H
at 4.8.5-4) compiled on Jul  3 2016 22:35:19 (with assert checking)
(1 row)
```

However, We cannot re-produce this issue on gpdb4.

```
[gpadmin@15d7534d2c68 regress]$ psql postgres -c 'select version();'
                                                                                    version

---------------------------------------------------------------------------------------------------------------
----------------------------------------------------------------
 PostgreSQL 8.2.15 (Greenplum Database 4.3.9.0 build dev) on x86_64-unknown-linux-gnu, compiled by GCC gcc (GCC
) 4.4.2 compiled on Jul  6 2016 04:19:33 (with assert checking)
(1 row)

[gpadmin@15d7534d2c68 regress]$ psql -p 65432 postgres -t -c 'select 1=1;'
 t
```

**Comment by ascherbaum-pivotal** [comment]
Hello @kuien, is this still an issue? Can you reproduce this without pgbouncer as well?

**Comment by hlinnaka** [comment]
This is a duplicate of issue #1001. I was able to reproduce it, after reverting the fix for #1001 (commit f94d9a7f0d1613c2c589d80f23806a0d31ce3c0b), but not on unmodified checkout from master. Using the auth_user option in pgbouncer was key to reproducing it, as that makes pgbouncer issue the auth_query using extended query protocol.

Closing.
